src/neuroconv/datainterfaces/ecephys/neuralynx/neuralynxdatainterface.py

```diff
@@ -118,3 +83,109 @@ def __init__(self, folder_path: FolderPathType, sampling_frequency: float = None
         """
 
         super().__init__(folder_path=folder_path, sampling_frequency=sampling_frequency, verbose=verbose)
+
+
+def get_filtering_multi_channel(neo_reader) -> List[str]:
+    """
+    Get the filtering metadata from neo reader containing a multiple channels.
+
+    Parameters
+    ----------
+    neo_reader: NeuralynxRawIO
+        Neo IO to extract the filtering metadata from
+
+    Returns
+    -------
+    list:
+        list of (str) json dump of filter parameters of each channel.
+         Uses the mu character, which may cause problems
+        for downstream things that expect ASCII.
+    """
+    neo_annotations = neo_reader.raw_annotations
+
+    # extracting infos of each signal (channel)
+    filter_info = []
+    stream_infos = neo_annotations["blocks"][0]["segments"][0]["signals"]
+    # iterate across streams
+    for stream_info in stream_infos:
+        stream_annotations = stream_info["__array_annotations__"]
+        filter_dict = {k: v for k, v in stream_annotations.items() if k.lower().startswith("dsp")}
+        # iterate across channels in stream
+        for chan_idx in range(len(stream_info["__array_annotations__"]["channel_names"])):
+            channel_filter_dict = {k: v[chan_idx] for k, v in filter_dict.items()}
+            # conversion to string values
+            for key, value in channel_filter_dict.items():
+                if not isinstance(value, str):
+                    value = " ".join(value)
+                channel_filter_dict[key] = value
+            filter_info.append(json.dumps(channel_filter_dict, ensure_ascii=True))
+
+    return filter_info
+
+
+def extract_metadata_single_reader(neo_reader) -> dict:
```

Comment:
```suggestion
def extract_neo_header_metadata(neo_reader) -> dict:
```

More specific, I guess that the single reader was for the previous case when multi readers were considered for the old backend.

Reply:
Done.